[Flaky Test] [sig-auth] TestPodSubresourceAuth - ci-kubernetes-integration-arm64-master

### Which jobs are flaking?
* sig-release-master-blocking
* integration-arm64-master

### Which tests are flaking?
* [k8s.io/kubernetes/test/integration/apiserver.apiserver](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-arm64-master/2038745793132761088)
* https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-arm64-master/2037835284841762816

### Since when has it been flaking?
* First flaky: Mon, 16 Mar 2026 07:02:56 UTC
* Latest flaky: Tue, 31 Mar 2026 05:32:19 UTC

### Testgrid link
* https://testgrid.k8s.io/sig-release-master-blocking#integration-arm64-master&exclude-non-failed-tests=
* [Triage](https://storage.googleapis.com/k8s-triage/index.html?text=expected%20nil%20error%20for%20user%20with%20%27create%27%20permissions%2C%20but%20got%3A%20pods%20%22test-pod%22%20is%20forbidden%3A%20User%20%22pod-creator%22&job=ci-kubernetes-integration-arm64-master%24&test=k8s.io%2Fkubernetes%2Ftest%2Fintegration%2Fapiserver%20apiserver)

### Reason for failure (if possible)
```
=== RUN   TestPodSubresourceAuth/subresource=exec
    subresource_auth_test.go:173: expected nil error for user with 'create' permissions, but got: pods "test-pod" is forbidden: User "pod-creator" cannot get resource "pods/exec" in API group "" in the namespace "test-pod-subresource-auth"
--- FAIL: TestPodSubresourceAuth/subresource=exec (0.00s)

=== RUN   TestPodSubresourceAuth/subresource=attach
    subresource_auth_test.go:173: expected nil error for user with 'create' permissions, but got: pods "test-pod" is forbidden: User "pod-creator" cannot get resource "pods/attach" in API group "" in the namespace "test-pod-subresource-auth"
--- FAIL: TestPodSubresourceAuth/subresource=attach (0.00s)

=== RUN   TestPodSubresourceAuth/subresource=portforward
    subresource_auth_test.go:173: expected nil error for user with 'create' permissions, but got: pods "test-pod" is forbidden: User "pod-creator" cannot get resource "pods/portforward" in API group "" in the namespace "test-pod-subresource-auth"
--- FAIL: TestPodSubresourceAuth/subresource=portforward (0.00s)

=== RUN   TestPodSubresourceAuth
--- FAIL: TestPodSubresourceAuth (3.28s)
```


### Anything else we need to know?
_No response_

### Relevant SIG(s)
/sig auth
/kind flake
cc @kubernetes/release-team-release-signal

--- a/test/integration/apiserver/subresource_auth_test.go
+++ b/test/integration/apiserver/subresource_auth_test.go
@@ -25,9 +25,12 @@ import (
 	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/test/integration/authutil"
 	"k8s.io/kubernetes/test/integration/framework"
 	"k8s.io/kubernetes/test/utils/ktesting"
 )
@@ -139,6 +142,11 @@ func TestPodSubresourceAuth(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// Wait for the permissions to become effective by ensuring a create request is allowed
+	waitContext, cancel := context.WithTimeout(context.TODO(), wait.ForeverTestTimeout)
+	defer cancel()
+	authutil.WaitForNamedAuthorizationUpdate(t, waitContext, adminClientset.AuthorizationV1(), podCreatorUsername, ns, "create", "", schema.GroupResource{Group: "", Resource: "pods/exec"}, true)
 
 	subresources := []string{"exec", "attach", "portforward"}
 	for _, subresource := range subresources {

--- a/test/integration/authutil/authutil.go
+++ b/test/integration/authutil/authutil.go
@@ -62,6 +62,7 @@ func WaitForNamedAuthorizationUpdate(t *testing.T, ctx context.Context, c author
 		}
 		return true, nil
 	}); err != nil {
+		t.Logf("did not see expected allowed response %v for user=%s,namespace=%s,verb=%s,resourceName=%s,resource=%v", allowed, user, namespace, verb, resourceName, resource)
 		t.Fatal(err)
 	}
 }

